waiter/src/waiter/state/responder.clj

```diff
@@ -681,11 +682,17 @@
                          release-instance-chan
                          (timers/start-stop-time!
                            responder-release-timer
-                           (handle-release-instance-request
-                             current-state service-id update-slot-state-fn update-status-tag-fn
-                             update-state-by-ejecting-instance-fn update-instance-id->eject-expiry-time-fn
-                             work-stealing-received-in-flight-counter max-eject-time-ms
-                             eject-backoff-base-time-ms max-backoff-exponent data))
+                           (let [[{instance-id :id} _] data
```

Comment:
Should we put this additional logic inside of `handle-release-instance-request`?  Isn't it part of the "handling release instance" process?

Reply:
I thought about it while implementing, but I like the way we have these helped `handle-*` functions structured. We do not have them do any side-effects (other than metrics), the side-effects are triggered in the go-blocks here. 

If you strongly prefer, I'm okay changing the code to move the logic to `handle-release-instance-request`